board/arm/nrf52xx_boards: NFCT pins configuration

nvlsianpu opened the pull request:
It is possible to use NFCT pins as regular GPIOs.
This patch introduce option for enabling this feature.

Fix for #6937 

Additionally this fix nrf52 part of #7452

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
arch/arm/soc/nordic_nrf/nrf52: NFCT pins configuration

It is possible to use NFCT pins as regular GPIOs.
This patch introduce option for enabling this feature.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

anangl commented on boards/arm/nrf52840_pca10056/Kconfig:
```diff
@@ -10,4 +10,15 @@ config GPIO_AS_PINRESET
 
+config NFCT_PINS_AS_GPIOS
+	bool "NFCT pins acts as GPIOs"
+	default n
```
There is no need to specify it, it is set implicitly.

anangl commented on boards/arm/nrf52840_pca10056/Kconfig:
```diff
@@ -10,4 +10,15 @@ config GPIO_AS_PINRESET
 	default y
 
+config NFCT_PINS_AS_GPIOS
+	bool "NFCT pins acts as GPIOs"
```
... pins act ...
Or maybe omit "act"?

anangl reviewed (commented):
What about pca10040?

carlescufi reviewed (approved):
Looks good.

anangl commented:
Shouldn't the same be done for pca10040 as well? I thought this PR would be a good place for this.

nvlsianpu committed:
boards/arm/nrf52xx_boards: makes GPIO_AS_PINRESET common

Kconfig keyword GPIO_AS_PINRESET is common for whole nrf52 series.
This patch move it to the series Kconfig.soc file.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu commented:
Initially I was thinking about adding NFCT_PINS_AS_GPIOS keyword to the arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc file, but options form this file are available in menuconfig via [ARM Options]->[SoC Options]->*. That why I add another Kconfig.board file.

galak commented on arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc:
```diff
@@ -31,7 +31,7 @@ endchoice
 config ARM_MPU_NRF52X
 	bool "Enable MPU on nRF52"
-	depends on CPU_HAS_MPU
+	depends on CPU_HAS_MPU && SOC_SERIES_NRF52X
```
Drop this change from this PR, should be in a separate PR.

galak reviewed (changes requested):
Drop the MPU change from this PR.

nvlsianpu commented:
@galak done

galak commented on arch/arm/soc/nordic_nrf/nrf52/Kconfig.board:
```diff
@@ -0,0 +1,21 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+config GPIO_AS_PINRESET
```
We should put these in  arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc as the code that implements them exists in arch/arm/soc/nordic_nrf/nrf52/soc.c

galak reviewed (changes requested):
Rather than introducing a new arch/arm/soc/nordic_nrf/nrf52/Kconfig.board and sourcing it from boards.  Lets put these Kconfig symbols in the Kconfig.soc as that is where the code that implements them exists.

nvlsianpu commented:
@galak I moved both GPIO_as_pin_reset and NFCT_as_GPIO options to the Kconfig.soc

Image below shows how it is looks like now:
![image](https://user-images.githubusercontent.com/17042885/40108600-abac8f5c-58fb-11e8-9349-6c76c34e15b8.png)
